extensions/amp-bind/0.1/bind-expr-impl.jison

```diff
@@ -2,44 +2,57 @@
 
 %{
 // Shortcuts for common functions.
-var toString = Object.prototype.toString;
-var hasOwnProperty = Object.prototype.hasOwnProperty;
+const toString = Object.prototype.toString;
+const hasOwnProperty = Object.prototype.hasOwnProperty;
 
 // For security reasons, must not contain functions that mutate the caller.
-var functionWhitelist =
-{
-  '[object Array]':
-    {
-      'concat': Array.prototype.concat,
-      'includes': Array.prototype.includes,
-      'indexOf': Array.prototype.indexOf,
-      'join': Array.prototype.join,
-      'lastIndexOf': Array.prototype.lastIndexOf,
-      'slice': Array.prototype.slice,
-    },
-  '[object String]':
-    {
-      'charAt': String.prototype.charAt,
-      'charCodeAt': String.prototype.charCodeAt,
-      'concat': String.prototype.concat,
-      'includes': String.prototype.includes,
-      'indexOf': String.prototype.indexOf,
-      'lastIndexOf': String.prototype.lastIndexOf,
-      'repeat': String.prototype.repeat,
-      'slice': String.prototype.slice,
-      'split': String.prototype.split,
-      'substr': String.prototype.substr,
-      'substring': String.prototype.substring,
-      'toLowerCase': String.prototype.toLowerCase,
-      'toUpperCase': String.prototype.toUpperCase,
-    },
-};
+const functionWhitelist = (() => {
+  const whitelist = {
+    '[object Array]':
+      [
+        Array.prototype.concat,
+        Array.prototype.includes,
+        Array.prototype.indexOf,
+        Array.prototype.join,
+        Array.prototype.lastIndexOf,
+        Array.prototype.slice,
+      ],
+    '[object String]':
+      [
+        String.prototype.charAt,
+        String.prototype.charCodeAt,
+        String.prototype.concat,
+        String.prototype.includes,
+        String.prototype.indexOf,
+        String.prototype.lastIndexOf,
+        String.prototype.repeat,
+        String.prototype.slice,
+        String.prototype.split,
+        String.prototype.substr,
+        String.prototype.substring,
+        String.prototype.toLowerCase,
+        String.prototype.toUpperCase,
+      ],
+  };
+  // Creates a prototype-less map of function name to the function itself.
+  // This makes function lookups faster (compared to Array.indexOf).
+  const out = Object.create(null);
+  Object.keys(whitelist).forEach(type => {
+    out[type] = Object.create(null);
+
+    const functions = whitelist[type];
+    for (let i = 0; i < functions.length; i++) {
+      const f = functions[i];
+      out[type][f.name] = f;
```

Comment:
The name property has spotty support in iOS and Android Browser (non chrome). Instead of that, we could use strings and grab the method off the prototype:

```js
const whitelist = [
  {
    class: '[object Array]',
    proto: Array.prototype,
    methods: ['concat', 'includes', ...]
  },
];